experimental/shared/src/main/scala/zio/prelude/experimental/EitherCompose.scala

```diff
@@ -44,6 +46,21 @@ object EitherCompose {
       case Right(b) => b2c(b)
     }
   }
+
+  implicit val URIOEitherCompose: EitherCompose[URIO] = new EitherCompose[URIO] {
```

Comment:
Same thing here. Can't all these instances be collapsed into one instance at the top of the hierarchy and then we don't have to keep reimplementing `compose`?